src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -1030,6 +1017,29 @@ private static void VerifyModelForOutVarInNotExecutableCode(SemanticModel model,
             Assert.Equal(expectedType, typeInfo.Type);
             Assert.Equal(expectedType, typeInfo.ConvertedType);
             Assert.Equal(typeInfo, ((CSharpSemanticModel)model).GetTypeInfo(decl));
+            Assert.True(model.GetConversion(decl).IsIdentity);
```

Comment:
This change is focused specifically on type syntax. You are suggesting to add tests on declaration expression syntax. Also, this test is correct in general in the context that it is supposed to be used, i.e. Out Variable Declarations. The scenario that you mention is not supposed to go through the test helper. Nevertheless, it would be interesting to have a test for that scenario, but it is clearly outside of scope for this change. I will open a follow-up issue for that.